lib/cli.js

```diff
@@ -1,28 +1,33 @@
-let doc = `
+const doc = `
 Usage:
   node-legofy <input> [<output>]
+  node-legofy
+
 Arguments:
-  <input>    Path to input image file.
-  <output>   Path to output image file.
+  <input>   Path to input image file.
+  <output>  Path to output image file [default: stdout if stdout is a file,
+            or '.lego' extention derived from input otherwise].
+
 Options:
-  -h, --help            Bring help.
-  --version             Show version.
-`;
+  -h, --help  Bring help.
+  --version   Show version.
+`
 
-import { docopt } from 'docopt';
-import Legofy from './index';
-import pkg from '../package.json';
+import { docopt } from 'docopt'
+import pipe from 'multipipe'
+import legofy from './index'
+import pkg from '../package.json'
 
-export default function cli (argv) {
-  let options = docopt(doc, {
-    version: pkg.version, argv: argv
-  });
+const elapsed = (start, end) => end - start
+const elapsedSeconds = (start, end) => elapsed(start, end) / 1000
+const display = seconds => console.error(`Image LEGOfied in ${seconds} seconds.`)
 
-  const time = new Date().valueOf();
+export default function cli (argv) {
+  console.error('')
+  const options = docopt(doc, { version: pkg.version, argv: argv })
+  const time = Date.now()
+  const end = () => display(elapsedSeconds(time, Date.now()))
 
-  new Legofy({
-    input: options['<input>'],
-    output: options['<output>'],
-    callback: () => console.log(`Image LEGOfied in ${(new Date().valueOf() - time) / 1000} seconds.`)
-  });
+  pipe(process.stdin, legofy().on('end', end), process.stdout)
```

Comment:
`legofy` is just a transform stream, taking stdin as input and outputing to stdout.

Also `end` stream event used to display the end message when the legofication is done.

Using `pipe` module for easy error handling of all streams instead of adding an `error` callback on each stream.